internal/pkg/cli/svc_init.go

```diff
@@ -180,23 +182,19 @@ func newInitSvcOpts(vars initSvcVars) (*initSvcOpts, error) {
 	return opts, nil
 }
 
-// Validate returns an error if the flag values passed by the user are invalid.
+// Validate returns an error for any invalid optional flags.
 func (o *initSvcOpts) Validate() error {
-	if o.appName == "" {
-		return errNoAppInWorkspace
-	}
-	if o.wkldType != "" {
-		if err := validateSvcType(o.wkldType); err != nil {
-			return err
+	if o.appName != "" {
```

Comment:
This looks good to me but do we need `wsAppName` as a member if `initSvcOpts`? Could we just
```
wsApp := tryReadingAppName()
if o.appName != "" {
}
o.appName = wsApp
```

Reply:
it's for the ease of testing